addons/web/static/src/views/pivot/pivot_renderer.js

```diff
@@ -103,11 +113,25 @@ export class PivotRenderer extends Component {
     /**
      * Handle a click on a header cell.
      *
+     * @param {PointerEvent} ev
      * @param {Object} cell
-     * @param {string} type col or row
+     * @param {boolean} isXAxis
      */
-    onHeaderClick(cell, type) {
-        if (cell.isLeaf && cell.isFolded) {
+    onHeaderClick(ev, cell, isXAxis) {
+        const type = isXAxis ? "col" : "row";
+        if (cell.isLeaf && !cell.isFolded) {
+            const target = ev.target.closest(".o_pivot_header_cell_closed");
+            Object.assign(this.dropdownReactive, {
+                target,
+                menuPosition: isXAxis ? "bottom-start" : "bottom-end",
+                customGroupBys: this.model.metaData.customGroupBys,
+                onItemSelected: (payload) =>
```

Comment:
put handlers outside and modify signatures? groupId and type would be parameters or retreived from state?